packages/docs/docs/tutorial/part3.md

```diff
@@ -0,0 +1,125 @@
+fantasty role-playing game, and then ensure that the output is formatted as either
+JSON or YAML.
+
+## Generating a plain-text character sheet
+
+Let's start with a component called `<CharacterGenerator>` that will generate a list of
+fields
+
+```tsx filename="packages/tutorial/src/part3.tsx"
+function CharacterGenerator() {
+  return (
+    <>
+      The following is a character profile for an RPG game in JSON format:{'\n'}
+      <CharacterField fieldName="name" />
+      {'\n'}
+      <CharacterField fieldName="class" />
+      {'\n'}
+      <CharacterField fieldName="race" />
+      {'\n'}
+      <CharacterField fieldName="list of no more than three weapons" />
+      {'\n'}
+      <CharacterField fieldName="list of no more than two spells" />
+    </>
+  );
+}
+```
+
+The component will generate text like:
+
+```
+The following is a character profile for an RPG game in JSON format:
+name: Aria
+class: Ranger
+race: Elf.
+list of no more than three weapons: Long sword, Short bow, Dagger.
+list of no more than two spells: Firebolt, Mage Armor.
+```
+
+where each field is generated in parallel by a component called `<CharacterField>`.
+
+## Generating one field of the character sheet
+
+```tsx filename="packages/tutorial/src/part3.tsx"
+function CharacterField(props: { fieldName: string }) {
+  return (
+    <>
+      {props.fieldName}:{' '}
+      <ChatCompletion>
+        <SystemMessage>
+          The user is generating a character sheet for a fantasy role-playing game. Your job is to provide a single
+          value for one of the fields of the character sheet. Please only return the chosen value, and no other
+          conversational text.
+        </SystemMessage>
+        <UserMessage>Please generate the character sheet field: {props.fieldName}</UserMessage>
+      </ChatCompletion>
+    </>
+  );
+}
+```
+
+`<CharacterField>` is simply a use of `<ChatCompletion>` to invoke the LLM. The
+`<SystemMessage>` component contains the system prompt to instruct the LLM in general
+how to respond, and the `<UserMessage>` component contains the specific prompt for this
+field in the character sheet.
+
+## Formatting the output as JSON and YAML
+
+Now, we're going to use two new components -- `<JsonChatCompletion>` and `<YamlChatCompletion>` --
+to take the plain-text output of `<CharacterGenerator>` and format it as JSON and YAML.
+
+```tsx filename="packages/tutorial/src/part3.tsx"
+const app = (
+  <Inline>
+    <CharacterGenerator />
+    {'\n\n'}
+    The following is a JSON representation of this character profile:{'\n'}
+    {(conversation) => (
+      <JsonChatCompletion>
+        <UserMessage>{conversation}</UserMessage>
+      </JsonChatCompletion>
+    )}
+    {'\n\n'}
+    And here is a YAML representation of this character profile:{'\n'}
+    {(conversation) => (
+      <YamlChatCompletion>
+        <UserMessage>{conversation}</UserMessage>
+      </YamlChatCompletion>
+    )}
+  </Inline>
+);
```

Comment:
I think I made a similar point elsewhere, but I wouldn't do this with `Inline`. I think it's far more natural to do something like:

```tsx
const character = memo(<CharacterGenerator />)
const app = (
  The following is a JSON representation of this character profile:{'\n'}
  <JsonChatCompletion>
    <UserMessage>{character}</UserMessage>
  </JsonChatCompletion>
  < YamlChatCompletion >
    <UserMessage>{character}</UserMessage>
  </YamlChatCompletion >
)
```

Using `Inline` is also a little funny because we're prompting the `Yaml` component with both the plaintext and JSON output, which seems unnecessary.